config/webpack.config.web2.js

```diff
@@ -40,18 +40,55 @@ async function makeConfig(
     devServer: {
       inline: false,
       before: (app /*: ExpressApp */) => {
-        const apiRoot = "/api/v1/data";
+        let developmentInstancePath /*: ?string */ =
+          process.env["SOURCECRED_DEV_INSTANCE"];
+        const lastArg = process.argv[process.argv.length - 1];
+        if (lastArg.startsWith("--instance")) {
+          const pieces = lastArg.split("=");
+          developmentInstancePath = pieces[1];
+        }
+        if (developmentInstancePath == null) {
+          throw new Error(
+            "Please provide a SourceCred cli2 instance, via $SOURCECRED_DEV_INSTANCE, or --instance=PATH"
+          );
+        }
+        const configPath = path.join(
+          developmentInstancePath,
+          "sourcecred.json"
+        );
+        if (!fs.existsSync(configPath)) {
+          // Sanity check; we won't try to verify that the right output files are there, since it may change and the
+          // frontend should have helpful error messages in that case. But if there's no sourcecred.json file, then
+          // it's likely the developer has made a mistake and would find a quick failure helpful.
+          throw new Error(
+            `Provided instance directory ${developmentInstancePath} doesn't have a sourcecred.json file`
+          );
+        }
+
+        const configContents = fs.readFileSync(configPath);
+        const serveConfig = (_unused_req, res /*: ExpressResponse */) => {
+          res.status(200).send(configContents);
+        };
```

Comment:
@topocount Notice a lot of similar code to your PR (https://github.com/sourcecred/sourcecred/pull/1861) here, might be worth consolidating in the future